### PR TITLE
[Fix] Loosen aleph-message dependency

### PR DESCRIPTION
**Pull request by MHHukiewitz**
Problem: Too strict aleph-message dependency

Solution: loosen it to accept compatible versions to 0.4.2

**EDIT:**
https://github.com/aleph-im/aleph-message/pull/91 fixed the backwards compatibility issue. `aleph-message` has been upgraded to 0.4.3 so these workflows pass again
> Another problem: Further problems arose from importing aleph-message~=0.4.2 due to some newer syntax that was not supported in Python 3.9 yet.
> 
> Another solution: EOL for Python 3.9 & out-of-the-box Ubuntu-20.04.
> 
> Also fixed deprecation warning for deprecated GH workflows that utilized EOL node.js v16 by using newer versions of those workflows.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -38,7 +38,7 @@ install_requires =
     eciespy>=0.3.13; python_version>="3.11"
     typing_extensions
     typer
-    aleph-message==0.4.1
+    aleph-message~=0.4.2
```
Comment (nesitor) [inline setup.cfg:41]:
```suggestion
    aleph-message~=0.4.1
```
This should be like that to accept `0.4.1` and `0.4.2` versions

**Comment by MHHukiewitz** [comment]
Rebased on #93

**Comment by hoh** [comment]
Commits mix different topics, can you ensure only the right commits are in this branch ?

**Review by nesitor — APPROVED** [review]

**Comment by MHHukiewitz** [comment]
Test is failing as aleph-message=0.4.2 has syntax that is not supported anymore in Python 3.8. Should we drop support for that version?

**Comment by hoh** [comment]
> Test is failing as aleph-message=0.4.2 has syntax that is not supported anymore in Python 3.8. Should we drop support for that version?

Yes, we can drop support for Python < 3.9

**Comment by MHHukiewitz** [comment]
I don't know why the new check for ubuntu 24.04 isn't being picked up on... @hoh

File: .github/workflows/build-wheels.yml [hunk 2]
```diff
@@ -12,24 +12,24 @@ jobs:
   build:
     strategy:
       matrix:
-        os: [macos-11, macos-12, ubuntu-20.04, ubuntu-22.04]
```
Comment (hoh) [inline .github/workflows/build-wheels.yml:15]: Can we keep support for Ubuntu 20.04 until it is EOL ? That is Apr 2025
See https://ubuntu.com/about/release-cycle

File: docker/ubuntu-24.04.dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:24.04
```
Comment (hoh) [inline docker/ubuntu-24.04.dockerfile:1]: Is that released already ?

File: .github/workflows/build-wheels.yml (hunk 2 above)
Comment (MHHukiewitz) [reply .github/workflows/build-wheels.yml]: Yes, reverted that and fixed by upgrading to patched aleph-message~=0.4.3

File: docker/ubuntu-24.04.dockerfile (hunk 3 above)
Comment (MHHukiewitz) [reply docker/ubuntu-24.04.dockerfile]: Reverted

**Review by hoh — APPROVED** [review]

**Pull request merged**